injection/sharedmain/main.go

```diff
@@ -90,12 +90,12 @@ func GetLoggingConfig(ctx context.Context) (*logging.Config, error) {
 	return logging.NewConfigFromConfigMap(loggingConfigMap)
 }
 
-func Main(component string, ctors ...injection.ControllerConstructor) {
+func Main(component, metricsName string, ctors ...injection.ControllerConstructor) {
```

Comment:
This is a backwards incompatible change.
What you probably should do instead is create a new `MainWithMetricName` and thread that through there.

Reply:
@vagababov yes you are right.